submissions/DmytryjK/js-oop-exercise-tiny/index.js

```diff
@@ -0,0 +1,96 @@
+/* Refer to https://github.com/OleksiyRudenko/a-tiny-JS-world for the task details
+   Complete the below for code reviewers' convenience:
+
+   Code repository: _put repo URL here_
+   Web app: _put project's github pages URL here_
+   */
+
+// ======== OBJECTS DEFINITIONS ========
+// Define your objects here
+
+
+// ======== OUTPUT ========
+/* Use print(message) for output.
+   Default tag for message is <pre>. Use print(message,'div') to change containing element tag.
+
+   Message can contain HTML markup. You may also tweak index.html and/or styles.css.
+   However, please, REFRAIN from improving visuals at least until your code is reviewed
+   so code reviewers might focus on a single file that is index.js.
+   */
+
+/* Print examples:
+   print('ABC');
+   print('<strong>ABC</strong>');
+   print('<strong>ABC</strong>', 'div');
+
+   print('human; John; male; 2; 2; Hello world!; Rex, Tom, Jenny');
+   print('human; <strong>John</strong>; male; 2; 2; <em>Hello world!</em>; Rex, Tom, Jenny');
+   print('human; <strong>John</strong>; male; 2; 2; <em>Hello world!</em>; Rex, Tom, Jenny', 'div');
+   */
+
+class Inhabitant {
+   constructor(name, spacies, gender, saying) {
+      this.name = name;
+      this.spacies = spacies;
+      this.gender = gender;
+      this.saying = saying;
+      this.property = ['name', 'spacies', 'gender', 'saying'];
+   }
+   get props() {
+      return this.property.map(key => this[key]).join('; ');
+   }
+}
+
+class Human extends Inhabitant {
+   constructor (name, gender, saying, spacies, legs, hands) {
+      super(name, spacies, gender, saying);
+      this.spacies = 'human';
+      this.legs = 2;
+      this.hands = 2;
+      this.property.push('legs', 'hands');
```

Comment:
We do not want a superclass to handle child class properties.
This breaks Encapsulation principle of OOP and Open-Closed principle of SOLID.
The language has tools to handle this situatuion.
Use method overloading and opportunity to call inherited method.